### PR TITLE
[WIP] Committer guidelines

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
CC @Andersson007 @gundalow

##### ISSUE TYPE
- Docs Pull Request
- Feature Pull Request

##### COMPONENT NAME
commit-rights.md

### CODE REVIEW

File: commit-rights.md
Comment (Andersson007) [inline commit-rights.md:4]:
```suggestion
This document is based on the [Ansible committer guidelines](https://github.com/ansible/ansible/blob/b57444af14062ec96e0af75fdfc2098c74fe2d9a/docs/docsite/rst/community/committer_guidelines.rst).
```
I hope my PR will get merged soon... and we can put the doc ref here

**Review by Andersson007 — COMMENTED** [review]
Very good idea, thanks!
I started to read it and I think it's worth waiting until the updated version is merged.
What do you think?

File: commit-rights.md
Comment (gundalow) [inline commit-rights.md:2]:
```suggestion
Committers Guidelines for community.general
===========================================
```

**Review by gundalow — COMMENTED** [review]
Do you want this as specific to c.g, or more general?

File: commit-rights.md
Comment (felixfontein) [reply commit-rights.md]: I assumed it is already specific enough since it is in the c.g repository - but you're right, it's better to be explicit :)

File: commit-rights.md
Comment (Andersson007) [inline commit-rights.md:25]:
```suggestion
Sometimes, committers merge their own pull requests. This section is a set of guidelines. If you are changing a comma in a doc or making a very minor change, you can use your best judgement. This is another trust thing. The process is critical for any major change, but for little things or getting something done quickly, use your best judgement and make sure people on the team are aware of your work.
```

File: commit-rights.md
Comment (Andersson007) [inline commit-rights.md:29]:
```suggestion
* Release managers: Merge pull requests to `stable-X` branches, create tags to do releases.
```
Same for all the following cases or, alternatively, we could create a section at the beginning deciphering the abbreviations

File: commit-rights.md
Comment (Andersson007) [inline commit-rights.md:35]:
```suggestion
Individuals with direct commit access to [ansible-core](https://github.com/ansible/ansible) are entrusted with powers that allow them to do a broad variety of things--probably more than we can write down. Rather than rules, treat these as general *guidelines*, individuals with this power are expected to use their best judgement.
```
Is it necessary here? Or do you mean `[community.general](...)`?

File: commit-rights.md
Comment (Andersson007) [inline commit-rights.md:37]:
```suggestion
* Do NOT
```

File: commit-rights.md
Comment (Andersson007) [inline commit-rights.md:42]:
```suggestion
  - Drag your community team members down. Always discuss the technical merits, but you should never address the person's limitations (you can later go for beers and call them idiots, but not in IRC/GitHub/and so on).
```
this style reminds me the Kernel style much:) Maybe a reference to CoC would suite here?

File: commit-rights.md
Comment (Andersson007) [reply commit-rights.md:37]: I'd also put "Do not .." at the beginning of each sentence and replace this one with something else

File: commit-rights.md
Comment (Andersson007) [inline commit-rights.md:53]:
```suggestion
  - Consider backwards compatibility (goes back to "don not break existing playbooks").
```

File: commit-rights.md
Comment (Andersson007) [inline commit-rights.md:66]:
```suggestion
Individuals who have been asked to become a part of this group have generally been contributing in significant ways to the community.general collection for some time. Should they agree, they are requested to add their names and GitHub IDs to this file, in the section below, through a pull request. Doing so indicates that these individuals agree to act in the ways that their fellow committers trust that they will act.
```

**Review by Andersson007 — COMMENTED** [review]
There were also several great suggestions by the other folks https://github.com/ansible/ansible/pull/73782

I think we shouldn't forget to add refs to CONTRIBUTING.md and REVIEW_CHECKLIST.md.
You could also add them yourself here (or ask me to do it), say, from https://github.com/ansible-collections/community.mysql where they have been already merged

File: commit-rights.md
Comment (felixfontein) [reply commit-rights.md:35]: Let's formulate it more generally:
```suggestion
Individuals with direct commit access to this collection repository are entrusted with powers that allow them to do a broad variety of things--probably more than we can write down. Rather than rules, treat these as general *guidelines*, individuals with this power are expected to use their best judgement.
```
What do you think?

File: commit-rights.md
Comment (felixfontein) [reply commit-rights.md]: Updated, thanks!

File: commit-rights.md
Comment (felixfontein) [reply commit-rights.md]: I've added a reference to our CoC.

File: commit-rights.md
Comment (aminvakil) [inline commit-rights.md:53]:
```suggestion
  - Consider backwards compatibility (goes back to "do not break existing playbooks").
```

File: commit-rights.md
Comment (aminvakil) [inline commit-rights.md:57]:
```suggestion
  - Consider scope, sometimes a fix can be generalized.
```

File: commit-rights.md
Comment (aminvakil) [inline commit-rights.md:41]: I had a hard time understanding this sentence and @felixfontein clarified it for me in #ansible-community IRC channel, but I don't know if there is something that can be added to make it more transparent or if it was just me :)

Comment (felixfontein) [reply commit-rights.md:41]: It should definitely be improved then. I'll think about it this evening :)

File: commit-rights.md
Comment (Andersson007) [inline commit-rights.md:4]:
```suggestion
This document is based on the [Ansible committer guidelines](https://docs.ansible.com/ansible/devel/community/committer_guidelines.html).
```
it has been updated finally:)

File: commit-rights.md
Comment (Andersson007) [reply commit-rights.md:21]: Looks good to me. What do you think about removing the section later when you add CONTRIBUTING.md and REVIEW_CHECKLIST.md?

File: commit-rights.md
Comment (Andersson007) [reply commit-rights.md:35]: Looks good tom me now:)

**Review by Andersson007 — COMMENTED** [review]
LGTM
would be good to clarify "other environments" a bit, so agreed with @aminvakil. Generally all the suggestions LGTM

File: commit-rights.md
Comment (aminvakil) [reply commit-rights.md:41]: Thanks!

File: commit-rights.md
Comment (felixfontein) [reply commit-rights.md:4]: I intentionally used the link to that commit, since that was the precise version of the document I copied everything from. The document you link to is a large refactoring :) How about:
```suggestion
This document is based on the [Ansible committer guidelines](https://github.com/ansible/ansible/blob/b57444af14062ec96e0af75fdfc2098c74fe2d9a/docs/docsite/rst/community/committer_guidelines.rst) ([latest version](https://docs.ansible.com/ansible/devel/community/committer_guidelines.html)).
```
?

File: commit-rights.md
Comment (felixfontein) [reply commit-rights.md:21]: Only if these documents also contain a short version of the workflow ;-) But let's see once we start adding them here.

File: commit-rights.md
Comment (felixfontein) [reply commit-rights.md:41]: @aminvakil what do you think of:
```suggestion
  - Do not forget about non-standard / alternate environments. Consider the alternatives. Yes, people have bad/unusual/strange environments (like binaries from multiple init systems installed), but they are the ones who need us the most.
```

Comment (aminvakil) [reply commit-rights.md:41]: I think this clarifies the message, but I'm still not sure if I can understand what do you mean about "binaries from multiple init systems installed" ?

Comment (felixfontein) [reply commit-rights.md:41]: Like having `systemctl` installed (from systemd) while you don't actually use systemd. If software like Ansible sees `systemctl` and thinks "this must mean that systemd is the init system", that assumption is wrong for this specific user.

Comment (aminvakil) [reply commit-rights.md:41]: That's what I was thinking, but who would have done that? In what purpose?

I'm not sure how many users do this, but is there any other non-standard environments which can be put here as an example?

I would not bother me handling a situation where user has not installed `systemd` but is using `systemctl` really :)

Comment (aminvakil) [reply commit-rights.md:41]: I'm not sure if these help or not for custom environments which are more used:

- another shell except bash
- dropping loopback in firewall
- custom linux kernels which do not have common flags enabled during build
- systems without a init by default like alpine

File: commit-rights.md
Comment (Andersson007) [reply commit-rights.md:4]: ah, clear now, the suggested version also works:)

File: commit-rights.md
Comment (felixfontein) [reply commit-rights.md:41]: > I'm not sure how many users do this, but is there any other non-standard environments which can be put here as an example which is more common?

Apparently it is quite common in containers (for some people at least?).

Comment (Andersson007) [reply commit-rights.md:41]: Can't we remove the sentence completely?
The initial one sounds a bit weird to me. Something like: "Don't forget about non-standard strange environments that is freaking hard but can be imagined." or "When merging, try to imagine a weird environment that the changes will break":)
I'd remove it at all...

Comment (felixfontein) [reply commit-rights.md:41]: I think we should keep something that describes the basic idea and not remove it completely.

Comment (Andersson007) [reply commit-rights.md:41]: Ok, maybe rephrase the initial one generalizing and simplifying

Comment (aminvakil) [reply commit-rights.md:41]: > Apparently it is quite common in containers (for some people at least?).

I wasn't aware of this, if that's common then it's fine to me, because as I've said before this clears the message that "beware of weird environments".

**Comment by felixfontein** [comment]
More comments/suggestions/...?

**Review by Andersson007 — APPROVED** [review]
LGTM

**Comment by Andersson007** [comment]
I'd add `often caused by bad vodka or something` to "strange environments" to add clarity but now it's also LGTM:)

**Comment by felixfontein** [comment]
@Andersson007 @gundalow @aminvakil thanks a lot for reviewing this!

**Pull request merged**
